src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/forwardinghandler.cpp

```diff
@@ -180,6 +180,10 @@ FORWARDING_HANDLER::ExecuteRequestHandler()
         if (cchHeader == 9 && _stricmp(pszWebSocketHeader, "websocket") == 0)
         {
             m_fWebSocketEnabled = TRUE;
+
+            // WinHttp does not support any extensions being returned by the server, so we remove the request header to avoid the server
+            // responding with any accepted extensions.
+            pRequest->DeleteHeader("Sec-WebSocket-Extensions");
```

Comment:
I'm assuming this is fine even if the header doesn't exist?

Reply:
The doc comments don't discuss it, but looking at the code it seems to check for existence internally. Also, the non-compression websocket tests should cover that scenario.